master/internal/api_project.go

```diff
@@ -520,6 +526,19 @@ func (a *apiServer) getProjectNumericMetricsRange(
 	return metricsValues, searcherMetricsValue, nil
 }
 
+func validateProjectKey(key string) error {
+	switch {
+	case len(key) > project.MaxProjectKeyLength:
+		return errors.Errorf("project key cannot be longer than %d characters", project.MaxProjectKeyLength)
+	case len(key) < 1:
+		return errors.New("project key cannot be empty")
+	case !regexp.MustCompile(ProjectKeyRegex).MatchString(key):
+		return errors.Errorf("project key can only contain alphanumeric characters")
+	default:
+		return nil
+	}
+}
+
 func (a *apiServer) PostProject(
```

Comment:
Do we want to add the ability to change the project key. Or do we want to add that in a separate ticket?